authentication/src/main/java/io/deephaven/auth/Privilege.java

```diff
@@ -0,0 +1,10 @@
+/*
+ * Copyright (c) 2016-2022 Deephaven Data Labs and Patent Pending
+ */
+package io.deephaven.auth;
+
+/**
+ * A privilege is a permission to perform a specific action. This is a token interface.
+ */
+public interface Privilege {
```

Comment:
It looks like everything that extends this must be an enum. If so, should define type:

`interface Privilege<T extends Privilege<T>> extends Enum<T>`

Reply:
Alternatively, you can update the builder method where it assumes they are enums.